pages/api/drb.ts

```diff
@@ -0,0 +1,47 @@
+import type { NextApiRequest, NextApiResponse } from "next"
+
+import type { SearchParams } from "../../src/types/searchTypes"
+import type { DRBResultsResponse } from "../../src/types/drbTypes"
+import nyplApiClient from "../../src/server/nyplApiClient/index"
+import { DRB_API_NAME } from "../../src/config/constants"
+import { getDRBQueryStringFromSearchParams } from "../../src/utils/drbUtils"
+import { mapQueryToSearchParams } from "../../src/utils/searchUtils"
+
+/**
+ * Getter function for DRB results. Accepts a SearchParams object as an argument.
+ */
+export async function fetchDRBResults(
+  searchParams: SearchParams
+): Promise<DRBResultsResponse | Error> {
+  const drbQueryString = getDRBQueryStringFromSearchParams(searchParams)
+
+  return nyplApiClient({ apiName: DRB_API_NAME })
+    .then((client) => client.get(drbQueryString))
```

Comment:
Apologies if there is already a bunch of code written in .then() format, but can we opt for async await for this entire repo?

Reply:
Yeah that's cool with me. Will fix in a bit.

Reply:
@charmingduchess I replaced the then statements with async/await in both drb.ts and search.ts, which cleaned it up substantially.

However, I think the Promise.all in search.ts looks a little cleaner with the .then syntax. How would you feel about keeping it in this case?

<img width="800" alt="Screenshot 2023-09-18 at 12 12 04 PM" src="https://github.com/NYPL/research-catalog/assets/6901105/0fcee005-bdd5-4553-bb2b-0dfef8421e15">

<img width="851" alt="Screenshot 2023-09-18 at 12 11 40 PM" src="https://github.com/NYPL/research-catalog/assets/6901105/2b831a67-f665-4598-97ec-d022d8340fb1">

Reply:
Is something missing from the second screenshot? Such as the `nyplApiClient` call and where `client` comes from?

Reply:
Yes, forgot to include that line in the screenshot, here it is with it:

<img width="830" alt="Screenshot 2023-09-18 at 7 14 24 PM" src="https://github.com/NYPL/research-catalog/assets/6901105/784f098a-c544-4035-a95e-21aa7c76d957">

Reply:
The more I look at it the more I kinda like the second option. Wouldn't it work with a `Promise.resolve(client.get(...))`? So you don't have to create the `new Promise(...)`?

Reply:
I like the sound of what Edwin is proposing. I'd rather not mix async await with .then's. I'm also a fan of this pattern, if it interests you (you may or may not need the async/await keywords here... not 100% on that syntax):
`Promise.all([resultsQuery, aggregationsQuery].map(async query => await client.get(${route}${query})`

But moreover, if the syntax of the first one works where you're awaiting promises, would this not work as well:
`client = await nyplClient()`
`Promise.all([await client.get(resultsPath), await client.get(aggsPath)]`

Reply:
Good call @charmingduchess ! Replaced with the syntax you suggestd in your last comment.